code/datums/diseases/advance/advance.dm

```diff
@@ -86,13 +86,16 @@ GLOBAL_LIST_INIT(advance_cures, list(
 
 // Compares type then ID.
 /datum/disease/advance/IsSame(datum/disease/advance/D)
+	if(ispath(D))
+		return FALSE
 
-	if(!(istype(D, /datum/disease/advance)))
-		return 0
+	if(!istype(D.type, /datum/disease/advance))
```

Comment:
we shouldnt be checking the .type var in istypes iirc, since its technically a path
```suggestion
	if(!istype(D, /datum/disease/advance))
```